### DISCUSSION TITLE
<Which routing package is suitable for my e-commerce app>

### DISCUSSION THREAD

**Discussion by BrillianceOsprey** [community/community#84738] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

1.Go Router
2.AutoRoute

**Comment by megamiii** [comment]
Go Router is ideal for simpler apps, where you might need basic navigation like switching between a home page, product list, and contact page. AutoRoute, however, suits more complex structures. For instance, if you have a product page with multiple categories and subcategories (nested routes), or need user authentication checks (guards) before accessing certain pages, AutoRoute would be more effective. Assess your app’s requirements to determine the best fit.

**Reply by BrillianceOsprey** [reply]
Thank you for your insightful comparison between Go Router and AutoRoute. Your breakdown of when each is ideal makes it much clearer for developers to assess their app's needs. I appreciate the guidance on choosing between them based on the complexity of the app's structure and the specific requirements such as nested routes and user authentication checks. It's a valuable reminder to always align the choice of routing library with the unique demands of the project. Great advice for navigating the route selection process!
